doc/source/reference/c-api.types-and-structures.rst

```diff
@@ -103,34 +104,38 @@ PyArray_Type and PyArrayObject
 
 .. c:member:: char *PyArrayObject.data
 
-    A pointer to the first element of the array. This pointer can
+    Accessible by the macro :c:data:`PyArray_DATA`, this data member is a
+    pointer to the first element of the array. This pointer can
```

Comment:
I like the link. Since it is not always a macro perhaps "Accessible via `:c:data:` ..." and add a general comment something like "Direct access to the structure fields is deprecated, use the `PyArray_*(arr)` form instead".

Reply:
I included this in the top paragraph